mempool/src/mempool.rs

```diff
@@ -132,7 +147,28 @@ impl Mempool {
     /// Stops the mempool executor
     ///
     /// This functions should only be called only after one of the functions to start the executor is called.
-    pub async fn stop_executor(&self) {
+    pub async fn stop_executor<N: Network>(&self, network: Arc<N>) {
+        let mut handle = self.executor_handle.lock().await;
+
+        if handle.is_none() {
+            // If there isn't any executor running we return
+            return;
+        }
+
+        // Unsuscribe to the network TX topic before killing the executor
+        network.unsubscribe::<TransactionTopic>().await.unwrap();
+
+        // Stop the executor
+        handle.take().expect("Expected an executor handle").abort();
+    }
+
+    /// Stops the mempool executor without TX stream
+    ///
+    /// This function is used for testing purposes (along with the start_executor_with_txn_stream function)
+    /// it is the responsability of the caller to suscribe and unsuscribe from the topic accordingly
+    ///
+    /// This functions should only be called only after one of the functions to start the executor is called.
+    pub async fn stop_executor_no_stream(&self) {
```

Comment:
I'd change the pattern to:
`subscribe_start_executor` and `unsusbscribe_stop_executor`
and
`start_executor` and `stop_executor`
Or something similar

Reply:
I like `start_executor` and `stop_executor` for the ones that are used in the code
For the testing ones, I would prefer:
`start_executor_with_txn_stream` and `stop_executor_without_txn_stream`

Reply:
Yeah, the one I'm not comfortable with is `stop_executor_without_txn_stream` since theoretically we do have at that point a `txn_stream`

Reply:
`stop_executor_without_unsuscribe` ?

Reply:
Sounds better to me since we won't unsubscribe as the name implies

Reply:
done